From ee777431a21c90bee6342db153b73a69ad1f14fb Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:50:00 -0400
Subject: [ethernet] bnx2x: Fix timesync endianity

Message-id: <1409946615-5862-56-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91608
O-Subject: [RHEL7.1 PATCH 55/70] bnx2x: Fix timesync endianity
Bugzilla: 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit 8f15c61301223138a01e62e009ad8b3a120b780c
Author: Michal Kalderon <Michal.Kalderon@qlogic.com>
Date:   Mon Aug 25 17:48:33 2014 +0300

    bnx2x: Fix timesync endianity

    Commit eeed018cbfa30 ("bnx2x: Add timestamping and PTP hardware clock support")
    has a missing conversion to LE32, which will prevent the feature from working
    on big endian machines.

    Signed-off-by: Michal Kalderon <Michal.Kalderon@qlogic.com>
    Signed-off-by: Yuval Mintz <Yuval.Mintz@qlogic.com>
    Signed-off-by: Ariel Elior <Ariel.Elior@qlogic.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sp.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sp.c
index 5fd8eca..954b35b 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sp.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sp.c
@@ -5877,8 +5877,10 @@ int bnx2x_func_send_set_timesync(struct bnx2x *bp,
   set_timesync_params->add_sub_drift_adjust_value;
  rdata->drift_adjust_value = set_timesync_params->drift_adjust_value;
  rdata->drift_adjust_period = set_timesync_params->drift_adjust_period;
- rdata->offset_delta.lo = U64_LO(set_timesync_params->offset_delta);
- rdata->offset_delta.hi = U64_HI(set_timesync_params->offset_delta);
+ rdata->offset_delta.lo =
+  cpu_to_le32(U64_LO(set_timesync_params->offset_delta));
+ rdata->offset_delta.hi =
+  cpu_to_le32(U64_HI(set_timesync_params->offset_delta));
 
  DP(BNX2X_MSG_SP, "Set timesync command params: drift_cmd = %d, offset_cmd = %d, add_sub_drift = %d, drift_val = %d, drift_period = %d, offset_lo = %d, offset_hi = %d\n",
     rdata->drift_adjust_cmd, rdata->offset_cmd,
-- 
1.7.1